library/java/net/openid/appauth/TokenRequest.java

```diff
@@ -185,6 +191,14 @@
     public final String scope;
 
     /**
+     * String value used to associate a Client session with an ID Token, and to mitigate replay attacks.
```

Comment:
Nonce is not sent to the token endpoint. Only the Authorization endpoint.  The client checks the nonce it sent against the nonce it received in the id_token 

The logic is that nonce allows the client to bind the request to a specific user agent.   In the native app case replay of a code from a different user agent is not the issue it is with a web server client.   However it is theoretically possable that  some script might try to inject a code from a different request into the response sent to the app.  In that case the id_token returned from the token endpoint would have a different nonce.    To be honest if you are doing PKCE that is a more recent standard it should tell you the same thing because the pkce verifier would be wrong.   

When using the 'code' response_type checking nonce is optional, and that is fine as long as you are doing PKCE with a view controller.   Without some way to cookie the custom tab from the app nonce is not really doing much.

John B.